src/GraphQL/Execution/ErrorParser.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace GraphQL.Execution
+{
+    public class ErrorParser : IErrorParser
+    {
+        private readonly ErrorParserOptions _options;
+
+        public ErrorParser()
+            : this(false)
+        {
+        }
+
+        public ErrorParser(bool exposeExceptions)
+            : this(new ErrorParserOptions() { ExposeExceptions = exposeExceptions })
+        {
+        }
+
+        public ErrorParser(ErrorParserOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public ErrorParser(Action<ErrorParserOptions> optionsBuilder)
+        {
+            if (optionsBuilder == null)
+                throw new ArgumentNullException(nameof(optionsBuilder));
+            _options = new ErrorParserOptions();
+            optionsBuilder(_options);
+        }
+
+        public ParsedError Parse(ExecutionError executionError)
+        {
+            IDictionary<string, object> extensions = null;
+            if (!string.IsNullOrWhiteSpace(executionError.Code) ||
+                // skip next check to match existing functionality
+                // (executionError.HasCodes) ||
```

Comment:
Currently, a new `ExecutionError` with an `InnerException` of `System.Exception` creates an `ExecutionError` with the `Code` property set to `String.Empty`.  This should be reviewed, as I would change it.  In any case, I was required to comment out the above line of code due to the semantics of the `HasCodes` property to pass all tests.